src/engine/LocalVocab.h

```diff
@@ -175,6 +181,22 @@ class LocalVocab {
   // generated by the blank node manager of this local vocab.
   bool isBlankNodeIndexContained(BlankNodeIndex blankNodeIndex) const;
 
+  // Wrapper around a bunch of word sets without any access provided to them.
+  // This is useful to extend the lifetime of this `LocalVocab` without making
+  // this instance read-only by merging and/or cloning.
+  class [[nodiscard(
+      "The sole purpose of this object is to extend lifetimes.")]] Holder {
+    std::vector<std::shared_ptr<const Set>> wordSets_;
+
+   public:
```

Comment:
This constructor can even be `private` (possibly with a friend if needed), s.t. it can only be created by the `getHolder` function inside the local vocab class.